processor/src/lib.rs

```diff
@@ -950,3 +950,27 @@ pub(crate) fn add_error_ctx_to_external_error(
         },
     }
 }
+
+/// Reads (start_addr, end_addr) tuple from the specified elements of the operand stack (
+/// without modifying the state of the stack), and verifies that memory range is valid.
+pub fn get_mem_addr_range(
+    process: &ProcessState,
+    start_idx: usize,
+    end_idx: usize,
+) -> Result<(u32, u32), MemoryError> {
```

Comment:
A potential small improvement: maybe this could be a method on the `ProcessState`? It could look something like:
```rust
impl ProcessState {
    pub fn get_mem_addr_range(
        &self,
        start_idx: usize,
        end_idx: usize,
    ) -> Result<Range<MemoryAddress>, MemoryError> {
        ...
    }
}
```
In the above, I've also used `Range<MemoryAddress>` instead of `(u32, u32)` for the return value as I think it is a bit more self-describing (and type-safe).

Reply:
Created a followup PR here: https://github.com/0xMiden/miden-vm/pull/2194